pkg/cmd/tests/options.go

```diff
@@ -15,18 +17,41 @@ type IngressControllerOptions struct {
 	CustomAnnotations map[string]string
 }
 
+type ScyllaClusterOptions struct {
+	NodeServiceType             string
+	NodesBroadcastAddressType   string
+	ClientsBroadcastAddressType string
+}
+
+var supportedNodeServiceTypes = []scyllav1.NodeServiceType{
+	scyllav1.NodeServiceTypeHeadless,
+	scyllav1.NodeServiceTypeClusterIP,
+}
+
+var supportedBroadcastAddressTypes = []scyllav1.BroadcastAddressType{
+	scyllav1.BroadcastAddressTypePodIP,
+	scyllav1.BroadcastAddressTypeServiceClusterIP,
+}
+
 type TestFrameworkOptions struct {
 	ArtifactsDir                 string
 	DeleteTestingNSPolicyUntyped string
 	DeleteTestingNSPolicy        framework.DeleteTestingNSPolicyType
 	IngressController            *IngressControllerOptions
+	ScyllaClusterOptionsUntyped  *ScyllaClusterOptions
+	scyllaClusterOptions         *framework.ScyllaClusterOptions
 }
 
 func NewTestFrameworkOptions() TestFrameworkOptions {
 	return TestFrameworkOptions{
 		ArtifactsDir:                 "",
 		DeleteTestingNSPolicyUntyped: string(framework.DeleteTestingNSPolicyAlways),
 		IngressController:            &IngressControllerOptions{},
+		ScyllaClusterOptionsUntyped: &ScyllaClusterOptions{
+			NodeServiceType:             string(scyllav1.NodeServiceTypeHeadless),
```

Comment:
Why headless? Normally, I'd tell people to use at least cluster IP to have discovery endpoint I assume we could use it in some tests at some point as well.

Reply:
I suppose this test the newest path, although it wouldn't be a production "default", so the answer would likely be we need the testing matrix eventually

Reply:
Now that you mentioned it I'm not sure it's that good of a default: it's the most "disruptive" in regard to some of our procedures, e.g. the ClusterIP-based replace procedure. On one hand it's a good thing, as it doesn't let any behaviour based on ClusterIPs slip (say e.g. the Scylla host for upgrades issue), on the other hand these test *will* fail with Headless services unless we run them conditionally.

Reply:
>  these test will fail with Headless services

which tests do you expect to fail on setups that don't use clusterIP? the CI looked green

Reply:
https://github.com/scylladb/scylla-operator/pull/1508#discussion_r1434219342 - the clusterIP-based replace tests fail without explicit override of exposeOptions (or other ways of circumventing it like skipping the tests or having sub-suites)
```
LAST SEEN   TYPE      REASON                   OBJECT                      MESSAGE
11s         Warning   InvalidScyllaDBVersion   scyllacluster/basic-25b5d   Can't proceed with the rollout. Specified ScyllaDB version "2022.2.12" does not support exposing via Service other than ClusterIP. Please use different ScyllaDB version.
```

Reply:
i thought we can just use newer scylladb there

Reply:
To my understanding the purpose of these tests is to specifically test the support for old procedure. After all we declare support for Scylla OS >= 5.0. The HostID-based procedure is only possible with >= 5.2.0.

Reply:
Where do our test use scylladb < 5.2.0? Manifest is fixed to `5.4.0` and upgrades start from `5.2.11`.

When we wire CI to run multiple scenarios (scylla versions) I expect we give it the right combination of scylla version and expose type to test.

Reply:
ok, I see the replace test has one more version spec - eventually the scylla version should be an argument but right now we mix it and that's probably why our reasoning here differs

Reply:
https://github.com/scylladb/scylla-operator/blob/master/test/e2e/set/scyllacluster/scyllacluster_replace.go#L186
https://github.com/scylladb/scylla-operator/blob/master/test/e2e/set/scyllacluster/scyllacluster_replace.go#L198

Reply:
And that's aligned with the test plan: https://github.com/scylladb/scylla-operator/tree/master/enhancements/proposals/1311-replace-nodes-using-hostid#test-plan.

Reply:
ok, let's skip the test and run a the other variant in a different job